### PR TITLE
[PersonaPlex] Capture the per-frame steps into CUDA graphs for PersonaPlexDepformer

### PR DESCRIPTION
**Pull request by HankBO**
## Purpose

This PR implements W2 of #7389: Depformer CUDA graph at fixed padded batch. It wraps `self.depformer` in `post_sample_talker_mtp` with a per-model graph wrapper keyed by padded B from cudagraph_capture_sizes, captured at warmup with padding rows, eager fallback on divergence logged once, following moss_tts/cuda_graph_streaming_decoder_wrapper.py:65-235(scratch slots for padding rows at :176-181, :207) and https://github.com/vllm-project/vllm-omni/pull/6604's approach. 

After unified duplex path is applied in `main`, it's not feasible to rebase dshah1333's work into this one. And the main task here is to re-target the design of #6604 into the unified duplex path only for Depformer. This PR will include dshah1333 as the co-author for credit sharing if still no response from #6604.

Default settings as below. the largest cuda graph size is 16, which is larger than `max_num_seqs: 10`. This makes sure cudagraph won't fallk back to eager model even with the largest batch.

```
    enforce_eager: false
    hf_overrides:
      depformer_cuda_graphs: true
    compilation_config:
      cudagraph_capture_sizes: [1, 2, 4, 8, 16]
      cudagraph_num_of_warmups: 3
```

## Implementation

- [x] Commit 1:
 - create `CUDAGraphDepformerWrapper` for capture, replay, and fallback.
 - **Static KV**: `_DepformerKVBuffers` is an `nn.Module` with `register_buffer` `k`/`v`, shape `[num_layers, max_batch, H, dep_q, Dh]`.
 - `warmup()` captures each size in `capture_sizes` (e.g. 1, 2, 4, 8). Replaly picks the smallest `padded_b >= actual_b`
 - Capture failure or no fitting size stays eager.
 - Use `DepformerCUDAGraphStats` to record states of cuda graph capturing and eager fall-back.
 - all the tests of `test_depformer.py`) and `test_depformer_cudagraph.py` passed.
 
`pytest -q -o addopts='' \
  tests/model_executor/models/personaplex/duplex/test_depformer.py \
  -m 'core_model and cpu' --run-level=core_model`

`pytest -q -o addopts='' \
  tests/model_executor/models/personaplex/duplex/test_depformer_cudagraph.py`

- [x] Commit 2: integrate `personaplex_talker.py`
 - integrate `CUDAGraphDepformerWrapper` into `post_sample_talker_mtp`.
 - add cuda-capture configs of hf_overrides in`vllm_omni/deploy/personaplex.yaml`
 - parse configs in `configuration_personaplex.py` to cotrol cuda-graph on/off
 - add testing in `test_depformer.py`

- [x] Bemchmark tesing: 

testing script: https://github.com/HankBO/vllm-omni/commit/193130a25a46ef60765870cf278ccbd92bd776d0

Reqeust:

```
python tests/e2e/online_serving/personaplex_realtime_duplex.py   --url ws://127.0.0.1:8099/v1/realtime?duplex=1   --model nvidia/personaplex-7b-v1   --input-wav tmp/input_assistant.wav  --output-dir tmp/personaplex-realtime-duplex
```

Service:

```
MODEL=<model_path>
nsys profile \
  --trace-fork-before-exec=true \
  -t cuda,nvtx,osrt \
  --capture-range=cudaProfilerApi \
  -o personaplex_main_b2_baseline \
  --force-overwrite=true \
python -m vllm_omni.entrypoints.cli.main serve   "$MODEL"   --omni   --deploy-config vllm_omni/deploy/personaplex.yaml   --host 0.0.0.0   --port 8099
```

**vLLM Version:** 0.29.0

**vLLM-Omni Commit:** 5a93ec1b4e990f42973954d63809f932e699426a

### Performance Benchmark

```
Baseline:
==================================================
 Nsys Profile Automated Analysis Report
==================================================
Total Trace Duration      : 8925.27 ms
Observed Anchor Ticks           : 100 (personaplex_sample_and_depformer)
Trace span / nominal tick : 111.6(metadata only; not observed tick count; nominal=80.0 ms)
--------------------------------------------------
Total Kernel records     : 450080
--------------------------------------------------
Absolute GPU Active Time  : 977.69 ms
Absolute GPU Idle Time    : 7947.58 ms
GPU Idle Fraction         : 89.05%
==================================================

==================================================
 NVTX Segment-Specific Kernel Analysis(launch-correlated)
==================================================
Segment (NVTX)       : personaplex_temporal_forward
Observed Occurrences  : 100 times
Total Kernels Records : 8
Total CUDA Runtime calls : 268
Total Graph Replays  : 100
Avg Correlated Kernels   : 0.1 kernels
Avg Graph Replays / Call: 1.0 replays
--------------------------------------------------
Segment (NVTX)       : personaplex_sample_and_depformer
Observed Occurrences  : 100 times
Total Kernels Records : 308288
Total CUDA Runtime calls : 661632
Total Graph Replays  : 0
Avg Correlated Kernels   : 3082.9 kernels
Avg Graph Replays / Call: 0.0 replays
--------------------------------------------------
```
```
Cuda-graph Depformer:
==================================================
 Nsys Profile Automated Analysis Report
==================================================
Total Trace Duration      : 3838.39 ms
Observed Anchor Ticks           : 100 (personaplex_sample_and_depformer)
Trace span / nominal tick : 48.0(metadata only; not observed tick count; nominal=80.0 ms)
--------------------------------------------------
Total Kernel records     : 144380
--------------------------------------------------
Absolute GPU Active Time  : 318.26 ms
Absolute GPU Idle Time    : 3520.14 ms
GPU Idle Fraction         : 91.71%
==================================================

==================================================
 NVTX Segment-Specific Kernel Analysis(launch-correlated)
==================================================
Segment (NVTX)       : personaplex_temporal_forward
Observed Occurrences  : 100 times
Total Kernels Records : 7
Total CUDA Runtime calls : 271
Total Graph Replays  : 100
Avg Correlated Kernels   : 0.1 kernels
Avg Graph Replays / Call: 1.0 replays
--------------------------------------------------
Segment (NVTX)       : personaplex_sample_and_depformer
Observed Occurrences  : 100 times
Total Kernels Records : 1000
Total CUDA Runtime calls : 7380
Total Graph Replays  : 100
Avg Correlated Kernels   : 10.0 kernels
Avg Graph Replays / Call: 1.0 replays
--------------------------------------------------
```

### Byte-comparison between eager vs. cudagraph

It seems that byte-identical cannot be retrieved through an e2e test with the unified duplex path. So, some metrics like cosine are used to measure the similarity using the script https://github.com/HankBO/vllm-omni/commit/4214b54bcfbcae49b0e4a9ba5cc48812c4deb932

```
{
  "note": "PCM similarity of one Realtime session, graphed depformer vs eager depformer. Not a byte-identical gate.",
  "metrics": {
    "eager_wav": "/tmp/personaplex-pcm-report/eager/primary-output.wav",
    "graphs_wav": "/tmp/personaplex-pcm-report/graphs/primary-output.wav",
    "sample_rate_hz": 24000,
    "eager_samples": 961920,
    "graphs_samples": 961920,
    "compared_samples": 961920,
    "length_delta_samples": 0,
    "cosine": -0.0002274307042122372,
    "max_abs_diff": 0.734375,
    "mean_abs_diff": 0.013570598562955062,
    "rms_diff": 0.03661290827130698,
    "byte_identical": false,
    "eager_peak": 0.474609375,
    "graphs_peak": 0.53125
  },
```

AI Usage Disclosure: GitHub Copilot is used to understand the context, code refinement, and cross-validation. All the code changes are reviewed and tested by the author.

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
This PR touches **tests/model_executor/, vllm_omni/model_executor/** (5 files). Based on CODEOWNERS coverage of the changed files, the most-related reviewers appear to be:

> @yenuo26 @NickCao @tzhouam

Could one of you take a look when you get a chance? Thanks!

**Comment by HankBO** [comment]
self-review:
- rebase the branch with #7413 
- use import current_omni_platform instead to avoid ci failure on AMD GPU
- do byte-comparison between eager vs. cudagraph e2e calling, receive a very close to zero cosinevalue.

**Comment by HankBO** [comment]
Hi @linyueqian , could you take a look this PR by any chance, thanks!
